docs/guides/tools.md

```diff
@@ -42,3 +42,66 @@ type Query {
   myName: String!
 }
 ```
+
+---
+
+### `depth_limit_validator`
+
+Create a validator to limit the complexity of queries by their depth to protect against malicious
+queries.
+
+```python
+from graphql import ValidationRule
+
+class DepthLimitOptions(TypedDict):
+    ignore: List[Union[str, re.Pattern, Callable[[str], bool]]]
+
+def depth_limit_validator(
+    max_depth: int,
+    options: Optional[DepthLimitOptions] = None,
+    callback: Optional[Callable[Dict[str, int]]] = None
+) -> ValidationRule:
```

Comment:
the way we use this below makes me think we could use a class instead of a function

Reply:
@patrick91 how can we use a function here? This is essentially a factory method to create a `ValidationRule` class.